src/Storages/ObjectStorage/DataLakes/Iceberg/IcebergMetadata.cpp

```diff
@@ -818,6 +831,7 @@ Strings IcebergMetadata::getDataFiles(const ActionsDAG * filter_dag) const
     for (const auto & manifest_list_entry : relevant_snapshot->manifest_list_entries)
     {
         auto manifest_file_ptr = getManifestFile(manifest_list_entry.manifest_file_path, manifest_list_entry.added_sequence_number);
+        initializeSchemasFromManifestFile(manifest_file_ptr);
```

Comment:
@divanik Why do we need this here in the first place? We don't use `schema_id_by_data_file`

AFAICS it will be called before by the `IcebergMetadata::getSchemaVersionByFileIfOutdated` `StorageObjectStorageSource::createReader`